src/linker/Linker/DocumentationSignatureParser.cs

```diff
@@ -500,7 +500,7 @@ static void GetMatchingTypes (ModuleDefinition module, TypeDefinition? declaring
 			Debug.Assert (module != null);
 
 			if (declaringType == null) {
-				var type = module.GetType (name);
+				var type = module.GetType (name) ?? module.ExportedTypes?.FirstOrDefault (et => et.FullName == name)?.Resolve ();
```

Comment:
Did you try constructing a TypeReference and resolving it instead? That feels like a more robust approach.
Also if we do keep this code - we should create a helper share it - since the doc parser below has exactly the same logic.

Specific case - I wonder how this works for nested types...